ergo-lib/src/chain/transaction.rs

```diff
@@ -55,7 +55,8 @@ use std::iter::FromIterator;
 )]
 #[derive(PartialEq, Debug, Clone)]
 pub struct Transaction {
-    tx_id: TxId,
+    /// transaction id
+    pub tx_id: TxId,
```

Comment:
The reason `tx_id` is private is that it's a computed property (hash of the serialized tx) and should not be set externally. Why do you want it to be public?